test/test_torch.py

```diff
@@ -6088,6 +6088,101 @@ def test_pow(self, device):
             torch.pow(m1, 1, out=out)
             self.assertEqual(out, m1)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @onlyOnCPUAndCUDA
+    @dtypes(*list(product(torch.testing.get_all_dtypes(include_bool=False, include_complex=False),
```

Comment:
Why are the tests with complex values separate?

Reply:
I merged this variant of test with the above. 
BTW, I might found another bug in original `pow()`. 
When `base` is a tensor with complexed 0, and `exponent` is a Scalar Tensor with 0 or a negative number, the results obtained by PyTorch would inconsistent with NumPy, just like below:
```
>>> import torch
>>> import numpy as np

>>> torch.tensor(0j)**torch.tensor(-1)
tensor(inf+nanj)
>>> np.array(0j)**np.array(-1)
(nan+nanj)
>>> torch.tensor(0j)**(-1)
tensor(nan+nanj)

>>> torch.tensor(0j)**torch.tensor(0)
tensor(nan+nanj)
>>> np.array(0j)**np.array(0)
(1+0j)
>>> torch.tensor(0j)**(0)
tensor(1.+0.j)
```
This may related with the similar issue #45690, and causes errors in my tests.
So I temporarily commented out the factors that may cause the error like:
```
base = make_tensor((30,), device, base_dtype, low=1, high=100)
# Complex and real results do not agree between PyTorch and NumPy when computing negative and zero power of 0
# Related: https://github.com/pytorch/pytorch/issues/45690
# base[0] = base[3] = base[7] = 0
exp = make_tensor((30,), device, exp_dtype, low=-2, high=2)
exp[0] = exp[4] = exp[6] = 0
```
And once this problem be solved, we can be restored it.

Reply:
Thank you for being so thorough and discovering this problem. Would you file a new issue for exactly this problem, too, and add a comment with a link to the new issue you file (instead of 45690)?

Reply:
Sure, a new issue for this problem -- #48000